xdem/coreg.py

```diff
@@ -947,12 +949,22 @@ def residuals(coefs: np.ndarray, x_coords: np.ndarray, y_coords: np.ndarray, tar
             print("Estimating deramp function...")
 
         # reduce number of elements for speed
-        if len(x_coords) > self.max_points:
-            indices = np.random.randint(0, len(x_coords) - 1, self.max_points)
+        # Get number of points to extract
+        max_points = len(x_coords)
```

Comment:
`len(arr)` is always equal to `arr.shape[0]`. Should it not be `.size`?
```python
In [1]: import numpy as np

In [2]: np.ones((4, 4))
Out[2]:
array([[1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.],
       [1., 1., 1., 1.]])

In [3]: len(np.ones((4, 4)))
Out[3]: 4

In [4]: np.ones((4, 4)).size
Out[4]: 16
```

Reply:
x_coords is a 1D array so it does not really matter here. But for clarity I will replace with size!

Reply:
Solved in 2ba3b5d